jaeger/spans.go

```diff
@@ -0,0 +1,121 @@
+package jaeger
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+	"sort"
+	"strconv"
+	"time"
+
+	jaegerModels "github.com/jaegertracing/jaeger/model/json"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/log"
+)
+
+// TODO / Question: make limit configurable? Selected from UI?
+const tracesLimit = 100
+
+type Span struct {
+	jaegerModels.Span
+	TraceSize int `json:"traceSize"`
+}
+
+func getSpans(client http.Client, endpoint *url.URL, namespace, service, startMicros, endMicros string) ([]Span, error) {
+	u := endpoint
+	queryService := service
+	if config.Get().ExternalServices.Tracing.NamespaceSelector {
+		queryService = service + "." + namespace
```

Comment:
Is this correct ?
I understood from some comments with @aljesusg we are going to fetch the "internal url" from config, instead to build the service using "namespace + selector" then we can reach same scenario but in and out cluster.

Reply:
ah, ok, nevermind. It adds the tracing namespace selector always ?

Reply:
To be honest I'm not 100% sure why this config of NamespaceSelector (a bool flag) exists, I guess someone asked for that at some point. This pattern of naming `service + "." + namespace` is what Envoy does when it creates traces, so that's why we need to do the same to identify service names as they're stored by Jaeger.

Reply:
Yes, I remember @aljesusg comments about it.

Reply:
Yes if namespaceSelector is true (by default) we look by service.namespace.

With false we look by service, there are users that not use the envoy to do this.